ExampleMod/Content/Projectiles/ExampleSineWaveMotionProjectile.cs

```diff
@@ -0,0 +1,150 @@
+			Projectile.ignoreWater = true; // Does the projectile's speed be influenced by water?
+			Projectile.light = 1f; // How much light emit around the projectile
+			Projectile.tileCollide = true; // Can the projectile collide with tiles?
+			Projectile.timeLeft = 600; // The live time for the projectile (60 = 1 second, so 600 is 10 seconds)
+		}
+
+		// This projectile updates its position manually
+		public override bool ShouldUpdatePosition() {
+			return false;
+		}
+
+		public override void OnSpawn(IEntitySource source) {
+			initialCenter = Projectile.Center;
+		}
+
+		public override Color? GetAlpha(Color lightColor) {
+			return drawColor * 0.64f;
+		}
+
+		public override void AI() {
+			// Ensure that the "waveDirection = 1" projectile starts above the "waveDirection = -1" projectile not matter which direction they were fired from
+			int trueDirection = Projectile.direction * waveDirection;
+
+			// How many oscillations happen per second
+			// Higher value = more oscillations
+			float wavesPerSecond = 2;
+
+			// Math.Sin expects a radians angle instead of degrees
+			float radians = MathHelper.ToRadians(sineTimer * 6f * wavesPerSecond);
+			float sine = (float)Math.Sin(radians) * trueDirection;
+
+			// Using the calculated sine value, generate an offset used to position the projectile on the wave
+			// The offset should be perpendicular to the velocity direction, hence the RotatedBy call
+			Vector2 offset = Projectile.velocity.SafeNormalize(Vector2.UnitX).RotatedBy(MathHelper.PiOver2 * -1);
+
+			// How wide the wave should be, times two
+			// An amplitude of 24 pixels is 1.5 tiles, meaning the total wave width is 48 pixels, or 3 tiles
+			float waveAmplitude = 24;
+
+			// Get the offset used to adjust the projectile's position
+			offset *= sine * waveAmplitude;
+
+			// Update the position manually since ShouldUpdatePosition returns false
+			initialCenter += Projectile.velocity;
+			Projectile.Center = initialCenter + offset;
+
+			// Update the rotation used to draw the projectile
+			// This projectile should act as if it were moving along the sine wave
+			float cosine = (float)Math.Cos(radians) * trueDirection;
+			Projectile.rotation = Projectile.velocity.ToRotation() + MathHelper.PiOver4 * cosine * -1;
+
+			// Update the frame used to draw the projectile
+			const float cosineOf45Degrees = 0.707106781f;
+			float cosineAbsolute = Math.Abs(cosine);
+			if (sine > 0) {
+				// Offset is counter-clockwise to the direction
+				if (cosineAbsolute > cosineOf45Degrees) {
+					Projectile.frame = Projectile.direction == 1 ? 0 : 2;
+				}
+				else {
+					Projectile.frame = 1;
+				}
+			}
+			else if (sine < 0) {
+				// Offset is clockwise to the direction
+				if (cosineAbsolute > cosineOf45Degrees) {
+					Projectile.frame = Projectile.direction == 1 ? 2 : 0;
+				}
+				else {
+					Projectile.frame = 1;
+				}
+			}
+			else {
+				// Offset is zero
+				Projectile.frame = 1;
+			}
```

Comment:
Comments are needed to explain more about the relationship between sprite and math here. My understanding is that you just want to change the sprite when you're at a 'peak' or 'trough' to indicate it'll be turning around.

You can detect those just by comparing `sine` though, and if I'm reading the math correctly, you currently have the 'curved' sprites (0 and 2) used when the projectile is travelling straight (0-45, 135-225, 315-... degrees)

Reply:
changed to only check `sine`